packages/framework-core/src/services/graphql/graphql-generator.ts

```diff
@@ -20,7 +21,20 @@ export class GraphQLGenerator {
   private readonly subscriptionGenerator: GraphQLSubscriptionGenerator
   private readonly typeInformer: GraphQLTypeInformer
 
-  public constructor(
+  private static singleton: GraphQLGenerator | undefined = undefined
+  
+  public static getInstance(config: BoosterConfig, logger: Logger): GraphQLGenerator {
+    if (this.singleton === undefined) {
+      this.singleton = new GraphQLGenerator(
+        config,
+        new BoosterCommandDispatcher(config, logger),
+        new BoosterReadModelDispatcher(config, logger)
+      )
+    }
+    return this.singleton;
+  }
```

Comment:
A simplification:

* Nullable fields are initializied to `undefined` by default
* You can use the `??` operator to return another value in case the left hand side operator of it is null or undefined

```suggestion
  private static singleton: GraphQLGenerator | undefined
  
  public static getInstance(config: BoosterConfig, logger: Logger): GraphQLGenerator {
    return this.singleton ?? new GraphQLGenerator(
      config,
      new BoosterCommandDispatcher(config, logger),
      new BoosterReadModelDispatcher(config, logger)
    )
  }
```